### PR TITLE
Fix logout for connected tests and re-enable Slack notifications

### PR DESCRIPTION
**Pull request by oguzkocer**
This PR adds an extra visibility matcher to `me_item` menu button in the "My Site" screen so the tests can logout. I couldn't figure out what caused the matcher to stop working since there doesn't seem to be a change to the `MySiteFragment` when the tests started to fail. Regardless, this matcher change is working locally, so hopefully it'll fix the ones in Firebase as well.

I've also re-enabled to Slack notifications since the tests were consistently passing before this issue, so assuming this PR fixes the issue, I don't think we need to keep that disabled anymore.

To test:
* Run the optional connected tests and verify all of them passes

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java [hunk 1]
```diff
@@ -33,7 +35,7 @@ public MePage go() {
         // Using the settings button as a marker for successfully navigating to the page
         while (!isElementDisplayed(appSettings)) {
             clickOn(R.id.nav_sites);
-            clickOn(R.id.me_item);
+            clickOn(onView(allOf(withId(R.id.me_item), withEffectiveVisibility(Visibility.VISIBLE))));
```
Comment (rachelmcr) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java:38]: Given that `clickOn` is our own custom click action, I don't love modifying the element we're passing it here. It feels cleaner to update the `clickOn` method itself to better handle scenarios like this. What do you think?

Comment (oguzkocer) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java:38]: Let's give it a shot.

Comment (oguzkocer) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java]: @rachelmcr I've changed the `onView(withId($elementId))` instances with a helper that also checks for visibility. This covers most of the matchers, but there are still some that checks for text or a specific child. I am not sure if we are doing the right thing by having the visibility matcher everywhere. That probably is due to my unfamiliarity with Espresso, so if you think we should change those as well, I can.

Comment (rachelmcr) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java]: > I am not sure if we are doing the right thing by having the visibility matcher everywhere.

Agreed. That helper feels off especially when combining `withEffectiveVisibility` and `isDisplayed` when it's used in various methods. I don't have as much experience with Espresso but my understanding is that those matchers are meant to be alternatives to each other, not additive checks (as explained in the [withEffectiveVisibility documentation](https://developer.android.com/reference/androidx/test/espresso/matcher/ViewMatchers#witheffectivevisibility).

I'm going to backtrack and suggest that we use your first approach, to limit how much we're using `withEffectiveVisibility` (only using it with elements that strictly require it). Sorry for going back and forth on that.

I'm also making a note to take another look at this behavior when I have time, to see if there's some other solution that I'm missing.

Comment (oguzkocer) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java]: @rachelmcr That's fine by me, but do you mind if we merge this PR and then change it back in a subsequent PR so the tests would work again? I don't have time to follow up with this today and might not be able to do it tomorrow either, so I don't want to postpone fixing the tests for too long.

Comment (rachelmcr) [reply WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/MePage.java]: Yes, I just went back and approved the PR before seeing this comment. I'll go ahead and merge it for now. 👍

**Pull request merged**